packages/req/src/index.ts

```diff
@@ -53,7 +53,9 @@ export const getFreshOrStale = (
 }
 
 export const checkIfXMLHttpRequest = (req: Pick<Request, 'headers'>): boolean =>
-  req.headers['X-Requested-With'] === 'XMLHttpRequest'
```

Comment:
header names are lowercase by default

https://nodejs.org/api/http.html#requestgetheaders

you can index by lowercase and it will work already, no need for a custom lookup